### ISSUE TITLE
[Bug]: Command failed icon when it was successful

### ISSUE THREAD

**Issue by malhotra5** [All-Hands-AI/OpenHands#5725]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Describe the bug and reproduction steps

NIT: The agent was running “grep” and didn’t find any matches

This was expected behavior but the frontend showed a failure icon - perhaps that’s OK but it confused me a little bit at first

### OpenHands Installation

Docker command in README

### OpenHands Version

saas

### Operating System

None

### Logs, Errors, Screenshots, and Additional Context

![IMG_4256](https://github.com/user-attachments/assets/2601c047-9f3c-4c22-904d-8ac738d8cee4)

**Comment by enyst** [comment]
I think the frontend is engaged in a guessing game, from the presence of the word "error" or so.

I haven't looked if we can do something now in this particular case. On the long run, I hope we can have more structured errors from the actions being run.

**Comment by mamoodi** [comment]
Yeah have definitely seen things like this for the error indication. Needs improvement.
